test/unit/config_test.rb

```diff
@@ -423,6 +459,40 @@
     end
   end
 
+  # ========================================================================== #
+  #                        Fetching Plugin Config
+  # ========================================================================== #
+  describe "when fetching plugin config" do
+    let(:cfg) { Inspec::Config.new({}, cfg_io) }
+    let(:cfg_io) { StringIO.new(ConfigTestHelper.fixture(fixture_name)) }
+    let(:fixture_name) { "basic_1_2" }
+
+    describe "when fetching a plugin config that is absent" do
+      let(:fixture_name) { "basic_1_2" }
+
+      it "returns an empty hash with indifferent access" do
+        settings = cfg.fetch_plugin_config("inspec-test-not-present")
+        refute_nil settings
+        assert settings.empty?
+        assert_kind_of Thor::CoreExt::HashWithIndifferentAccess, settings
+      end
+    end
+
+    describe "when fetching a plugin config that is present" do
+      let(:fixture_name) { "basic_1_2" }
+
+      it "returns the settings as a hash with indifferent access" do
+        settings = cfg.fetch_plugin_config("inspec-test-plugin")
+        refute_nil settings
+        refute settings.empty?
+        assert_kind_of Thor::CoreExt::HashWithIndifferentAccess, settings
+        assert_equal settings[:test_key_01], "test_value_01"
+        assert_equal settings["test_key_01"], "test_value_01"
+      end
```

Comment:
I actually prefer assertions over expectations. Mike Moore calls the combo of `describe/it` and assertions "assert spec style" and I think it can have a nice balance. This PR uses both "assert spec style" and "spec style". Any reason for the mix? (not asking for you to change it)

Reply:
I don't think anyone on the InSpec project set out with a "testing philosophy" in mind. Different files and different blocks within files use different approaches as different people worked on them, or as they learned new techniques, or as new styles became championed or outlawed.  Sheer pragmatism rules here: the code must work, and the tests must give some assurance of that, and be able to help us diagnose it when it the code fails.  I don't know what the styles are called, or even that we are mixing styles. A talk on that might be interesting....

Reply:
I'd be down to do that